File: src/main/java/arcana/utils/InventoryUtils.java
package arcana.utils;

import net.minecraft.inventory.InventoryHelper;
import net.minecraft.item.ItemStack;
import net.minecraft.nbt.CompoundNBT;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.World;
import net.minecraftforge.items.IItemHandler;

import java.util.Objects;

public class InventoryUtils {
    public static void dropContents(World world, BlockPos pos, IItemHandler items){
        int size = items.getSlots();
        for (int i = 0; i < size; i++){
            InventoryHelper.dropItemStack(world, pos.getX(), pos.getY(), pos.getZ(), items.getStackInSlot(i));
        }
    }

    public static void clear(IItemHandler items){
        int size = items.getSlots();
        for (int i = 0; i < size; i++){
            items.extractItem(i, items.getStackInSlot(i).getCount(), false);
        }
    }

    public static boolean stacksMatch(ItemStack a, ItemStack b){
        return a.getItem() == b.getItem() && tagsMatch(a.getTag(), b.getTag());
    }

    public static boolean tagsMatch(CompoundNBT a, CompoundNBT b){
        return Objects.equals(a, b);
    }
}

File: src/main/kotlin/arcana/utils/InventoryUtils.kt
package arcana.utils

import net.minecraft.inventory.InventoryHelper
import net.minecraft.item.ItemStack
import net.minecraft.nbt.CompoundNBT
import net.minecraft.util.math.BlockPos
import net.minecraft.world.World
import net.minecraftforge.items.IItemHandler
import java.util.*

object InventoryUtils {
    fun dropContents(world: World, pos: BlockPos, items: IItemHandler) {
        val size = items.slots
        for (i in 0 until size) {
            InventoryHelper.dropItemStack(world, pos.x.toDouble(), pos.y.toDouble(), pos.z.toDouble(), items.getStackInSlot(i))
        }
    }

    fun clear(items: IItemHandler) {
        val size = items.slots
        for (i in 0 until size) {
            items.extractItem(i, items.getStackInSlot(i).count, false)
        }
    }

    fun stacksMatch(a: ItemStack, b: ItemStack): Boolean {
        return a.item === b.item && tagsMatch(a.tag, b.tag)
    }

    fun tagsMatch(a: CompoundNBT?, b: CompoundNBT?): Boolean {
        return Objects.equals(a, b)
    }
}